### PR TITLE
fix: fix deepcopy torchtensor

### PR DESCRIPTION
**Pull request by JoanFM**

### CODE REVIEW

File: docarray/typing/tensor/torch_tensor.py [hunk 1]
```diff
@@ -271,3 +271,10 @@ def _docarray_from_ndarray(cls: Type[T], value: np.ndarray) -> T:
     def _docarray_to_ndarray(self) -> np.ndarray:
         """cast itself to a numpy array"""
         return self.detach().cpu().numpy()
+
+    def new_empty(self, *args, **kwargs):
```
Comment (samsja) [inline docarray/typing/tensor/torch_tensor.py:275]: can we copy the full signature of the original method ?

Comment (JoanFM) [reply docarray/typing/tensor/torch_tensor.py:275]: if it is always the same, I do not think it makes sense, this is way more mantainable

Comment (samsja) [reply docarray/typing/tensor/torch_tensor.py:275]: But this break everything related to mypy and pycharm feature. In DocArray v2 we always repeat the full signature of the function

Comment (JoanFM) [reply docarray/typing/tensor/torch_tensor.py:275]: but mypy check passes

Comment (JoanFM) [reply docarray/typing/tensor/torch_tensor.py:275]: The method on TorchEmbedding does the same.
This is a method that noone should use
It is easier to forget to update this method than anything else.

Comment (samsja) [reply docarray/typing/tensor/torch_tensor.py:275]: okay oaky

Comment (samsja) [reply docarray/typing/tensor/torch_tensor.py:275]: mypy pass because it does not look at it. It will only check if there is type hint

**Pull request merged**